ResearchKit/ActiveTasks/ORKSpeechRecognitionStep.m

```diff
@@ -87,7 +87,7 @@ - (BOOL)isEqual:(id)object {
     
     __typeof(self) castObject = object;
     return (isParentSame &&
-            (self.speechRecognizerLocale == castObject.speechRecognizerLocale));
+            [self.speechRecognizerLocale isEqual:castObject.speechRecognizerLocale]);
```

Comment:
This equality check wasn't working properly, it did pointer equality instead of value equality.